File: tests/Generator/resource/rust/complex/definition_type.rs
use serde::{Serialize, Deserialize};
use array_definition_type::ArrayDefinitionType;
use map_definition_type::MapDefinitionType;
use struct_definition_type::StructDefinitionType;

// Base definition type
#[derive(Serialize, Deserialize)]
pub struct DefinitionType {
    #[serde(rename = "deprecated")]
    deprecated: Option<bool>,

    #[serde(rename = "description")]
    description: Option<String>,

    #[serde(rename = "type")]
    _type: Option<String>,

}

File: tests/Generator/resource/kotlin/complex/DefinitionType.kt

import com.fasterxml.jackson.annotation.*

/**
 * Base definition type
 */
open abstract class DefinitionType {
    @JsonProperty("deprecated") var deprecated: Boolean? = null
    @JsonProperty("description") var description: String? = null
    @JsonProperty("type") var type: String? = null
}

